app/models/agency.rb

```diff
@@ -21,6 +21,11 @@ class Agency < ApplicationRecord
 
   scope :active, -> { where(status_id: 1) }
 
+  scope :with_event_on, lambda { |date|
+    joins(:event_dates)
+      .where('event_dates.event_date_key = ?', date)
+  }
+
   scope :with_event_after, lambda { |date|
```

Comment:
similar comment on scope name.  Could change to 'with_event_on_or_after'

Reply:
here this scope pulls only events on date (ex: 2020-10-10) so i just added name as `with_event_on`